test/unit/QueueConsumer-test.js

```diff
@@ -0,0 +1,314 @@
+    })
+  }
+
+  ack (message) {
+    return Promise.resolve()
+  }
+
+  nack (message) {
+    return Promise.resolve()
+  }
+
+  cancel (tag) {
+    return Promise.resolve()
+  }
+
+  close () {
+    if (this._closed) {
+      throw new Error('Already closed')
+    }
+    this._closed = true
+  }
+}
+
+class MockConnection extends EventEmitter {
+  constructor (channel) {
+    super()
+    this._mockChannel = channel
+  }
+
+  createChannel () {
+    return Promise.resolve(this._mockChannel)
+  }
+}
+
+const testQueueName = 'some-queue'
+const testMessage = { foo: 'bar' }
+
+test.beforeEach('setup mock channel and connections', (t) => {
+  const channel = new MockChannel(testQueueName)
+  const connection = new MockConnection(channel)
+  const consumer = new QueueConsumer({
+    queueName: testQueueName,
+    connection
+  })
+  t.context = { channel, connection, consumer }
+})
+
+test('should generate a _consumerTag based off of the queue name', (t) => {
+  const { consumer } = t.context
+  const consumerTag = consumer._consumerTag
+
+  t.true(consumerTag !== null)
+  t.true(consumerTag.startsWith(testQueueName))
+})
+
+/**
+ * Starting the queue consumer
+ */
+test('should should create a channel using the given connection', async (t) => {
+  const { channel, connection, consumer } = t.context
+  const mock = sinon.mock(connection)
+
+  mock.expects('createChannel').once()
+    .returns(channel)
+
+  await consumer.start()
+
+  mock.verify()
+  t.is(consumer._channel, connection._mockChannel)
+})
+
+test('should perform an assertion on the queue', async (t) => {
+  const { channel, consumer } = t.context
+  const mock = sinon.mock(channel)
+
+  mock.expects('assertQueue').once()
+    .withArgs(testQueueName)
+
+  await consumer.start()
+
+  mock.verify()
+  t.pass()
+})
+
+test('should set the channel prefetch', async (t) => {
+  const { channel, consumer } = t.context
+  const mock = sinon.mock(channel)
+
+  mock.expects('prefetch').once()
+    .withArgs(consumer._prefetchCount)
+
+  await consumer.start()
+
+  mock.verify()
+  t.pass()
+})
+
+test('should begin consuming from the queue', async (t) => {
+  const { channel, consumer } = t.context
+  const mock = sinon.mock(channel)
+
+  mock.expects('consume').once()
+    .withArgs(testQueueName)
+
+  await consumer.start()
+
+  mock.verify()
+  t.pass()
+})
+
+test('should emit a message if a message is consumed', async (t) => {
+  const { channel, consumer } = t.context
+
+  const messagePromise = new Promise((resolve) => {
+    consumer.once('message', (message) => {
+      resolve(message)
+    })
+  })
+
+  await consumer.start()
+
+  const testMessage = { msg: 'foo' }
+
+  channel.emit('test-message', testMessage)
+  let receivedMessage = await messagePromise
+
+  t.deepEqual(receivedMessage, testMessage)
+})
+
+test('should fail to consume if a channel cannot be made', async (t) => {
+  const { connection, consumer } = t.context
+  const mock = sinon.mock(connection)
```

Comment:
This mock never gets restored. I swear I didn't look at this test before making the suggestion above :)